format/format_test.go

```diff
@@ -443,6 +443,33 @@ a[_x[y][[z, w]]]`,
 			}(),
 			expected: `concat(",", [__local1__], __local0__)`,
 		},
+		{
```

Comment:
For this kind of change, the golden file tests seem sufficient. The other tests here usually have some specific thing that can't be easily asserted using just input and formatted output.

Reply:
I don’t mind that at all, but the only reason I found out about that file was from the race condition (!?) tests failing.. although there’s no obvious change that would affect concurrency that I could see. To simplify, I do not fully understand how they’re run, nor if they run as part of make go-build? I didn’t see the same errors on my machine, maybe due to Wasm being disabled?

Reply:
Race detection is a special test run. So if a test fails, it'll also fail the race detector test run. 

The test file pairs are read here, https://github.com/open-policy-agent/opa/blob/main/format/format_test.go#L80